tools/evict-shared-cache/evict-shared-cache.cpp

```diff
@@ -49,11 +50,50 @@ std::unique_ptr<EvictionPolicy> make_policy(const char* argv0, const char* polic
     return std::make_unique<NilEvictionPolicy>();
   }
 
-  std::cout << "Unknown policy: " << policy << std::endl;
+  std::cerr << "Unknown policy: " << policy << std::endl;
   print_help(argv0);
   exit(EXIT_FAILURE);
 }
 
+std::string read_command() {
+  std::string command = "";
+
+  // TODO: no idea if this is the right condition
+  while (std::cin.good() && std::cout.good()) {
```

Comment:
I think this is fine but if you really wanted to be paranoid you should use `fcntl(F_GETFD)` and see if it errors out with `EBADF` in which case the file handle is closed for sure.